JetAnalyzers/bin/jet_response_fitter_x.cc

```diff
@@ -255,6 +271,584 @@ int main(int argc,char**argv)
+    if( ks>1e-4 && fabs(ks-1)>0.1) mapKsFitM[ks]=hrspClone; 
+    if(chi2<50) mapChi2FitM[chi2]=hrspClone;
+
+    ////////////////
+    // fit case 1
+    ////////////////
+    modified_fdscb->SetLineColor(kGreen); 
+    for(int i=16; i<=17; i++) modified_fdscb->FixParameter(i, 0); // gaus2 right
+    modified_fdscb->FixParameter(18, 1); // sigma !=0 
+    hrsp->Fit("modified_fdscb", "R"); 
+    testfit = (TF1*) hrsp->GetListOfFunctions()->Last();
+    testfit->SetNpx(hrsp->GetNbinsX());
+    ks=hrsp->KolmogorovTest(testfit->GetHistogram()); 
+    chi2=testfit->GetChisquare()/max(1, testfit->GetNDF()); 
+    cout<<"ks/chi2 case1 "<<ks<<"/"<<chi2<<endl;
+    hrspClone = (TH1F*) hrsp->Clone(); 
+    if( ks>1e-4 && fabs(ks-1)>0.1) mapKsFitG[ks]=hrspClone; 
+    if(chi2<50) mapChi2FitG[chi2]=hrspClone;
+
+  } // random
+  cout<<"endRandom\n";
+  cout<<"retriving best fit from ks or chi2 map..."<<endl;
+
+  if(mapKsFitB.size()!=0)mapKsFit[mapKsFitB.rbegin()->first]=mapKsFitB.rbegin()->second; 
+  if(mapKsFitO.size()!=0)mapKsFit[mapKsFitO.rbegin()->first]=mapKsFitO.rbegin()->second; 
+  if(mapKsFitM.size()!=0)mapKsFit[mapKsFitM.rbegin()->first]=mapKsFitM.rbegin()->second; 
+  if(mapKsFitG.size()!=0)mapKsFit[mapKsFitG.rbegin()->first]=mapKsFitG.rbegin()->second; 
+  //
+  if(mapChi2FitB.size()!=0)mapChi2Fit[mapChi2FitB.begin()->first]=mapChi2FitB.begin()->second; 
+  if(mapChi2FitO.size()!=0)mapChi2Fit[mapChi2FitO.begin()->first]=mapChi2FitO.begin()->second; 
+  if(mapChi2FitM.size()!=0)mapChi2Fit[mapChi2FitM.begin()->first]=mapChi2FitM.begin()->second; 
+  if(mapChi2FitG.size()!=0)mapChi2Fit[mapChi2FitG.begin()->first]=mapChi2FitG.begin()->second; 
+
+  bool goodFit{false};
+
+  if(mapKsFit.size()!=0){
+    double maxKs{mapKsFit.rbegin()->first};
+    if(maxKs<0.5) { 
+      goodFit=false; hrspOrig= (TH1F*) mapKsFit.rbegin()->second->Clone();
+      cout<<"failed Ks: "<<mapKsFit.rbegin()->first<<endl;
+    }
+    else {
+      goodFit=true; hrspOrig= (TH1F*) mapKsFit.rbegin()->second->Clone();
+      cout<<"pass Ks: "<<mapKsFit.rbegin()->first<<endl;
+    }
+  }
+  else if(mapChi2Fit.size()!=0){
+      double minChi2{mapChi2Fit.begin()->first};
+      if(minChi2>5) { 
+	goodFit=false; hrspOrig= (TH1F*) mapChi2Fit.begin()->second->Clone();
+	cout<<"failed Chi2: "<<mapChi2Fit.begin()->first<<endl;
+      }
+      else {
+	goodFit=true; hrspOrig= (TH1F*) mapChi2Fit.begin()->second->Clone();
+	cout<<"pass Chi2: "<<mapChi2Fit.begin()->first<<endl;
+      }
+  }
+  else{ goodFit=false; cout<<"no fit at all, keeping original histogram!!\n"; }
+
+
+  // save fit plot
+  TCanvas *c = new TCanvas("c", "", 700, 700);
+  c->Divide(1,2);
+  c->cd(1); hrspOrig->Draw();
+  c->cd(2); gPad->SetLogy(); hrspOrig->Draw();
+
+  string savePath;
+
+  if( goodFit){
+    if(!calibrated) savePath="/home/calpas/JetMETAnalysis/CMSSW_7_4_15_patch1/src/JetMETAnalysis/JetAnalyzers/test/fitPlots/uncalibrated/"+alg+"/pass/"+hrspOrig->GetName()+".png";
```

Comment:
@Calpas please find a way to do this without referencing your own directory or any directories that aren't already included in the package. Maybe a command line option?